File: bindings/python/tests/test_liquidz.py
"""Tests for the Liquidz Python binding."""

import pytest
from liquidz import render, render_string, RenderError


def test_simple_variable():
    result = render("Hello, {{ name }}!", {"name": "World"})
    assert result == "Hello, World!"


def test_loop():
    result = render("{% for item in items %}{{ item }} {% endfor %}", {"items": ["a", "b", "c"]})
    assert result == "a b c "


def test_conditional():
    result = render("{% if show %}visible{% else %}hidden{% endif %}", {"show": True})
    assert result == "visible"


def test_filter():
    result = render("{{ name | upcase }}", {"name": "hello"})
    assert result == "HELLO"


def test_json_string_data():
    result = render("{{ x }}", '{"x": 42}')
    assert result == "42"


def test_empty_data():
    result = render("Hello")
    assert result == "Hello"


def test_none_data():
    result = render("Hello", None)
    assert result == "Hello"


def test_nested_objects():
    result = render("{{ user.name }}", {"user": {"name": "Alice"}})
    assert result == "Alice"


def test_render_string_alias():
    result = render_string("Hello, {{ name }}!", {"name": "World"})
    assert result == "Hello, World!"


def test_array_filters():
    result = render("{{ items | join: ', ' }}", {"items": ["a", "b", "c"]})
    assert result == "a, b, c"


def test_math_filters():
    result = render("{{ 5 | plus: 3 }}")
    assert result == "8"


def test_multiple_variables():
    result = render("{{ first }} {{ last }}", {"first": "John", "last": "Doe"})
    assert result == "John Doe"


def test_unless_tag():
    result = render("{% unless hide %}visible{% endunless %}", {"hide": False})
    assert result == "visible"


def test_assign_tag():
    result = render("{% assign x = 5 %}{{ x }}")
    assert result == "5"


def test_capture_tag():
    result = render("{% capture greeting %}Hello{% endcapture %}{{ greeting }}, World!")
    assert result == "Hello, World!"

File: bindings/ruby/test/test_liquidz.rb
require "minitest/autorun"
require "liquidz_ext"

class TestLiquidz < Minitest::Test
  def test_simple_variable
    result = Liquidz.render("Hello, {{ name }}!", { name: "World" })
    assert_equal "Hello, World!", result
  end

  def test_loop
    result = Liquidz.render("{% for item in items %}{{ item }} {% endfor %}", { items: ["a", "b", "c"] })
    assert_equal "a b c ", result
  end

  def test_conditional
    result = Liquidz.render("{% if show %}visible{% else %}hidden{% endif %}", { show: true })
    assert_equal "visible", result
  end

  def test_filter
    result = Liquidz.render("{{ name | upcase }}", { name: "hello" })
    assert_equal "HELLO", result
  end

  def test_json_string_data
    result = Liquidz.render("{{ x }}", '{"x": 42}')
    assert_equal "42", result
  end

  def test_empty_data
    result = Liquidz.render("Hello", nil)
    assert_equal "Hello", result
  end

  def test_nested_objects
    result = Liquidz.render("{{ user.name }}", { user: { name: "Alice" } })
    assert_equal "Alice", result
  end

  def test_render_string_alias
    result = Liquidz.render_string("Hello, {{ name }}!", { name: "World" })
    assert_equal "Hello, World!", result
  end
end
